cloud/deploy/deploy.go

```diff
@@ -879,8 +974,16 @@ func DeployClientImage(deployInput InputClientDeploy) error { //nolint:gocritic
 			fmt.Println("Building client image for host platform")
 		}
 
+		// Prepare build context with client dependency files
+		buildContext, err := prepareClientBuildContext(deployInput.Path)
+		if err != nil {
+			return fmt.Errorf("failed to prepare client build context: %w", err)
+		}
+		defer buildContext.CleanupFunc()
```

Comment:
```suggestion
		defer buildContext.CleanupFunc()
		if err != nil {
			return fmt.Errorf("failed to prepare client build context: %w", err)
		}
```
nit: to call the cleanup even in the error case

Reply:
See https://github.com/astronomer/astro-cli/pull/1980#discussion_r2486234556. This cleanup here is purely for the success case, where the build context has been set up correctly, but we clean up after the image build or when the image build itself fails.